### DISCUSSION TITLE
Random crash while using a Clickonce application with CefSharp 124.3.80

### DISCUSSION THREAD

**Discussion by yefka** [cefsharp/CefSharp#4818] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

124.3.80

### Operating System

Windows 11

### Architecture

x64

### .Net Version

.Net 4.7.2

### Implementation

WinForms

### Reproduction Steps

Since I upgraded from 117.* to 124.3.80 I get some random crash while starting the application.
No .Net exception is raised, and on the Debug console I just get an error message "Access Violation".
I've joined here a screenshot of the EventLog.

![image](https://github.com/cefsharp/CefSharp/assets/11507108/f1b79e0c-40b9-4969-981a-9445687d46dc)

The errors occurs on the application startup, during CefSharp initialization.
I only got this behavior while using a Clickonce application, it didn't occur with the 117.* version and below.

### Expected behavior

It should not crash.

### Actual behavior

About 90% of the time, the app crashes on startup

### Regression?

Yes

### Does this problem also occur in the CEF Sample Application

Yes using WPF/OffScreen command line args

**Comment by yefka** [comment]
I've realized that it occurs only when application is started twice.
The first one to start is ok.
The second one can't initialize Cef.Initialize(settings) returns false.

How can I diagnose what is wrong ?

**Comment by yefka** [comment]
I have confirmed that it's related to the Cache folder, it's not related to Clickonce.

If I start one instance then change the cache folder with that : 
`                settings.CachePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "Whatever", "Cache");`

Then the second one can start.

If I tried to run a 3rd one with the same setting, it crashes.
Cache folder can't be shared anymore by several instance of the app ?

**Reply by yefka** [reply]
Seems related to https://github.com/cefsharp/CefSharp/issues/4668

**Comment by yefka** [comment]
Workaround : 

```
            _cefSharpRootCache = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "MyApp", "Cache", Guid.NewGuid().ToString() /* hack, as Chromium does not support cache sharing anymore */);

            settings.RootCachePath = _cefSharpRootCache;
```

And when the application closes : 
```
            Cef.Shutdown();
            if (!string.IsNullOrWhiteSpace(_cefSharpRootCache) && Directory.Exists(_cefSharpRootCache))
                Directory.Delete(_cefSharpRootCache, recursive: true);
```

**Reply by amaitland** [reply]
Firstly it's important to note that if `Cef.Initialzie` returns **false** then it's not possible to create `ChromiumWebBrowser` instances

Using a unique cache path is obviously one option.

Supporting the process singleton is another. `IBrowserProcessHandler.OnAlreadyRunningAppRelaunch` will be called for in the first process, allowing you to launch a new window or handle the behaviour in a way appropriate to your application.

Launching multiple windows within the same process is how `Chromium` works.
